bundle/src/main/java/com/adobe/acs/tools/explain_query/impl/ExplainQueryServlet.java

```diff
@@ -181,16 +196,27 @@ protected final void doPost(SlingHttpServletRequest request, SlingHttpServletRes
         final Session session = resourceResolver.adaptTo(Session.class);
 
         try {
+            final QueryLogCollector logCollector = new QueryLogCollector(this.loggers, this.pattern,
```

Comment:
Try to create a new `QueryLogCollector` for each execution; registerCollector(..) will only allow 1 to be registered at a time thought.